src/disclosure-manager/_DisclosureContainer.jsx

```diff
@@ -45,26 +51,34 @@ const DisclosureContainer = injectIntl(({ intl, children, navigationPromptResolu
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
   useEffect(() => {
-    disclosureManager.registerDismissCheck(() => new Promise((resolve, reject) => {
-      if (!promptCheckpointRef.current) {
-        resolve();
-        return;
+    disclosureManager.registerDismissCheck(() => {
+      if (customRegisterDismissCheckRef.current) {
+        return customRegisterDismissCheckRef.current();
```

Comment:
Should this be 
```suggestion
        return customRegisterDismissCheckRef.current;
```
or should this execute and then return?

Reply:
Yep you got it, returning the Promise so that we can chain off it.